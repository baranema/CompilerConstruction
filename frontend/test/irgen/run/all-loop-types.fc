extern int printf(char[] format, ...);

int main() {
    int i = 3;

    printf("do while loop:\n");
    do {
        printf("%d\n", i);
        i--;
    } while(i > 0);

    printf("while loop:\n");
    while(i < 5){
        printf("%d\n", i);
        i++;
    }

    printf("for loop:\n");
    for(int j = 0 to 2){
        printf("%d\n", j);
    }

    printf("done\n");
    return 0;
}
