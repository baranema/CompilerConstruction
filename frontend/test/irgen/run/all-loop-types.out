do while loop:
3
2
1
while loop:
0
1
2
3
4
for loop:
1
2
done
